pkg/armadaevents/events.proto

```diff
@@ -476,4 +474,19 @@ message SeqUpdates{
 
 message DatabaseSequence{
     EventSequence eventSequence = 1;
-}
\ No newline at end of file
+}
+
+message JobPreempted {
```

Comment:
This should be `JobRunPreempted`, not `JobPreempted`. Since it's a specific run of the job that was preempted. Further, instead of adding all these fields at the top level of the message, we should have a oneof field. This oneof field would have as an option `PodPreempted`, which would contain the fields currently in the `JobPreempted` message. We do something similar for job submitted messages. This is to support preempting jobs that don't consist only of a single field in the future.

Reply:
I thought about this. Do we really need any more than this?
For now, we'd need to use the job id to populate the run id field.
```
message JobRunPreempted {
    // Id of the run that was preempted.
    Uuid preempted_run_id = 1;
    // Id of the job run that caused the preemption.
    Uuid preemptive_run_id = 2;
}
```